linux/src/async_socket_linux.c

```diff
@@ -406,12 +411,22 @@ static int initialize_global_thread(void)
                 result = -1;
                 (void)close(g_epoll);
                 g_epoll = -1;
-                interlocked_decrement(&g_thread_access_cnt);
+                (void)interlocked_decrement(&g_thread_access_cnt);
             }
         }
+        if (interlocked_decrement(&g_epoll_access) == 0)
```

Comment:
>g_epoll_access [](http://example.com/codeflow?start=35&length=14)

So, what happens if we delete g_epoll_access completely?
We should talk about this, I have a very uneasy feeling  introducing it when I don't understand really why we need it.